src/CustomElementRegistry.js

```diff
@@ -57,20 +57,39 @@ export default class CustomElementRegistry {
   }
 
   /**
-   * @param {string} localName
+   * @param {string} name
    * @param {!Function} constructor
    */
-  define(localName, constructor) {
+  define(name, constructor, options) {
     if (!(constructor instanceof Function)) {
       throw new TypeError('Custom element constructors must be functions.');
     }
 
-    if (!Utilities.isValidCustomElementName(localName)) {
-      throw new SyntaxError(`The element name '${localName}' is not valid.`);
+    if (!Utilities.isValidCustomElementName(name)) {
+      throw new SyntaxError(`The element name '${name}' is not valid.`);
     }
 
-    if (this._internals.localNameToDefinition(localName)) {
-      throw new Error(`A custom element with name '${localName}' has already been defined.`);
+    if (this._internals.nameToDefinition(name)) {
+      throw new Error(`A custom element with name '${name}' has already been defined.`);
+    }
+
+    let localName = name;
+
+    if (options && options.extends) {
```

Comment:
See `6.` and `7.` in [the spec for define](https://html.spec.whatwg.org/multipage/scripting.html#customelementregistry)